synapse/lib/cache.py

```diff
@@ -27,6 +27,8 @@ def __len__(self):
         return len(self.cache)
 
     def pop(self, key):
+        if key in self.fifo:
```

Comment:
Is it possible that this is not getting tested? Seems like the relative coverage drop may be just here.

Reply:
I think we originally "decided" not to delete from the fifo due to O(N) iteration concerns, but in retrospect that was probably dangerous due to potential GC side effects.  If we made the `self.cache.pop()` default to `s_common.novalu` and then only did the `self.fifo.remove()` in the event that it is not `novalu` we could probably split the difference. ( of course, remembering to return None if it was `novalu` to maintain backward compat )

Blind/unconfirmed `pop()` use cases should be minimal, but we can make an easy optimization.

Reply:
Good call. I did some performance testing and `if key in self.fifo` was an order of magnitude slower. The implementation that you suggested (that just got committed) is comparable to the original implementation.